tests/Unit/MetadataHydratorTest.php

```diff
@@ -283,15 +283,19 @@ public function testHydrateWithNormalizerInBaseClass(): void
                 ProfileId::fromString('1'),
                 Email::fromString('info@patchlevel.de'),
             ),
-            ['foo'],
+            [StatusWithNormalizer::Draft],
+            [StatusWithNormalizer::Draft],
+            [StatusWithNormalizer::Draft],
         );
 
         $event = $this->hydrator->hydrate(
             NormalizerInBaseClassDefinedDto::class,
             [
                 'status' => 'draft',
                 'profileCreated' => ['profileId' => '1', 'email' => 'info@patchlevel.de'],
-                'array' => ['foo'],
+                'defaultArray' => ['draft'],
+                'listArray' => ['draft'],
+                'iterableArray' => ['draft'],
```

Comment:
i think we should also add testcases resolving other sub-types besides enum. Like a "normal" object, a nested array with object etc.